### PR TITLE
Split of isotp.py

### PR DESCRIPTION
**Pull request by polybassa**
Replacement for #3079

### CODE REVIEW

**Comment by polybassa** [comment]
This PR doesn't contain any code changes, I just have split up isotp.py into multiple files to make this component more clean.

File: scapy/contrib/isotp/isotp_native_socket.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+CAN_ISOTP_DEFAULT_RECV_WFTMAX = 0
+CAN_ISOTP_DEFAULT_LL_MTU = CAN_MTU
+CAN_ISOTP_DEFAULT_LL_TX_DL = CAN_MAX_DLEN
+CAN_ISOTP_DEFAULT_LL_TX_FLAGS = 0
+
+
+class SOCKADDR(ctypes.Structure):
+    # See /usr/include/i386-linux-gnu/bits/socket.h for original struct
+    _fields_ = [("sa_family", ctypes.c_uint16),
+                ("sa_data", ctypes.c_char * 14)]
```
Comment (gpotter2) [inline scapy/contrib/isotp/isotp_native_socket.py:68]: This is far from standardized. See https://github.com/secdev/scapy/blob/100b2ca7d01128e059330d34f11d4b01d2744c7e/scapy/libs/winpcapy.py#L71

File: scapy/contrib/isotp/isotp_scan.py [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+                                                        return_values,
+                                                        noise_ids, True,
+                                                        pkt, verbose),
+                           timeout=sniff_time * 2,
+                           started_callback=lambda: sock.send(pkt))
+
+    return return_values
+
+
+def ISOTPScan(sock,  # type: SuperSocket
```
Comment (gpotter2) [inline scapy/contrib/isotp/isotp_scan.py:269]: Functions that start with caps are misleading if you ask me

File: scapy/contrib/isotp/isotp_soft_socket.py [hunk 3]
```diff
@@ -0,0 +1,1065 @@
...
+        self.exiting = False
+        self._thread_started = Event()
+        self.exception = None  # type: Optional[Exception]
+
+        Thread.__init__(self)
+        self.name = "CANReceiver" + self.name
+
+    def start(self):
+        # type: () -> None
+        Thread.start(self)
```
Comment (gpotter2) [inline scapy/contrib/isotp/isotp_soft_socket.py:278]: super is cleaner

File: scapy/contrib/isotp/isotp_soft_socket.py [hunk 4]
```diff
@@ -0,0 +1,1065 @@
...
+                try:
+                    callback()
+                except Exception:
+                    traceback.print_exc()
+
+    @staticmethod
+    def _time():
+        # type: () -> float
+        if six.PY2:
+            return time.time()
```
Comment (gpotter2) [inline scapy/contrib/isotp/isotp_soft_socket.py:492]: `time.monitonic()` equivalent on python 2 would be `time.clock()`

File: scapy/contrib/isotp/isotp_soft_socket.py [hunk 5]
```diff
@@ -0,0 +1,1065 @@
...
+                if isinstance(self._cb, bool):
+                    # Handle was already executed.
+                    # We don't need to cancel anymore
+                    return False
+                else:
+                    self._cb = None
+                    TimeoutScheduler.cancel(self)
+                    return True
+
+        def __cmp__(self, other):
```
Comment (gpotter2) [inline scapy/contrib/isotp/isotp_soft_socket.py:525]: Why do you need `__cmp__`?

File: scapy/contrib/isotp/isotp_soft_socket.py (hunk 3 above)
Comment (polybassa) [reply scapy/contrib/isotp/isotp_soft_socket.py:278]: Thanks!

File: scapy/contrib/isotp/isotp_soft_socket.py (hunk 4 above)
Comment (polybassa) [reply scapy/contrib/isotp/isotp_soft_socket.py:492]: thanks!

File: scapy/contrib/isotp/isotp_soft_socket.py (hunk 5 above)
Comment (polybassa) [reply scapy/contrib/isotp/isotp_soft_socket.py:525]: I don't need it, actually. Maybe it was a left over. Removed it. Thanks!

File: scapy/contrib/isotp/isotp_native_socket.py (hunk 1 above)
Comment (polybassa) [reply scapy/contrib/isotp/isotp_native_socket.py:68]: I've updated the names. actually SOCKADDR wasn't used in my code.. Removed it! Thanks

File: scapy/contrib/isotp/isotp_scan.py (hunk 2 above)
Comment (polybassa) [reply scapy/contrib/isotp/isotp_scan.py]: fixed!

**Comment by polybassa** [comment]
@gpotter2 updated. hopefully ready to merge, now.

**Comment by gpotter2** [comment]
Could you please rebase against master? Thanks

**Comment by polybassa** [comment]
@gpotter2 Rebased

**Comment by polybassa** [comment]
Thanks for merging

**Pull request merged**